public/app/features/dashboard/dashgrid/PanelHeader/PanelHeaderTitleItems.tsx

```diff
@@ -47,6 +54,21 @@ export function PanelHeaderTitleItems(props: Props) {
     </>
   );
 
+  const message = `This ${
+    angularNotice?.isAngularPanel
+      ? 'panel'
+      : angularNotice?.isAngularDatasource
+      ? 'data source'
+      : 'panel or data source'
+  } requires Angular (deprecated).`;
```

Comment:
I would probably extract this, for readability, into a function that does early return instead of nesting ternary operators.